src/stapi_fastapi/backends/product_backend.py

```diff
@@ -9,30 +9,42 @@
 from stapi_fastapi.models.order import Order, OrderPayload
 from stapi_fastapi.routers.product_router import ProductRouter
 
-
-class ProductBackend(Protocol):  # pragma: nocover
-    async def search_opportunities(
-        self,
-        product_router: ProductRouter,
-        search: OpportunityRequest,
-        request: Request,
-    ) -> ResultE[list[Opportunity]]:
-        """
-        Search for ordering opportunities for the  given search parameters.
-
-        Backends must validate search constraints and return
-        `stapi_fastapi.exceptions.ConstraintsException` if not valid.
-        """
-
-    async def create_order(
-        self,
-        product_router: ProductRouter,
-        search: OrderPayload,
-        request: Request,
-    ) -> ResultE[Order]:
-        """
-        Create a new order.
-
-        Backends must validate order payload and return
-        `stapi_fastapi.exceptions.ConstraintsException` if not valid.
```

Comment:
@jkeifer This statement isn't clear to me:
- Is it intending to say that order payloads must be validated against the product constraints? Raising a ConstraintsException would then make sense to me. 
- Or do we need an `OrderPayloadException`?

For now, I've regurgitated the existing wording.

Reply:
`ConstraintsException` might not be the best name anymore? I'm not actually 100% certain of the initial naming and how it correlated with the use of this exception, and it if ever was more accurately named, but regardless, I think you are right, we probably want a better named exception here. Something more generic would likely make more sense. I'm not sure `OrderPayloadException` is it either, I think we just rename `ConstraintsException` to be like `InvalidRequestError` or something.

I'd probably not change it here, and just make an issue for us to consider if this exception class is appropriately named given the usage.